toolkit/tools/imagegen/attendedinstaller/primitives/enumfield/enumfield.go

```diff
@@ -0,0 +1,252 @@
+		textColor:                tview.Styles.PrimaryTextColor,
+		textColorActivated:       tview.Styles.PrimaryTextColor,
+		options:                  options,
+	}
+}
+
+// SetLabelColor sets the color of button text.
+func (n *EnumField) SetLabelColor(color tcell.Color) *EnumField {
+	n.labelColor = color
+	return n
+}
+
+// SetLabelColorActivated sets the color of button text when the button is
+// selected.
+func (n *EnumField) SetLabelColorActivated(color tcell.Color) *EnumField {
+	n.labelColorActivated = color
+	return n
+}
+
+// SetNavBackgroundColor sets the background color.
+func (n *EnumField) SetFieldBackgroundColor(color tcell.Color) *EnumField {
+	n.backgroundColor = color
+	return n
+}
+
+// SetBackgroundColorActivated sets the background color of the button text when
+// the button is selected.
+func (n *EnumField) SetBackgroundColorActivated(color tcell.Color) *EnumField {
+	n.backgroundColorActivated = color
+	return n
+}
+
+// Draw renders this primitive onto the screen.
+func (n *EnumField) Draw(screen tcell.Screen) {
+
+	// Update dimensions
+	n.Box.Draw(screen)
+
+	// Obtain dimensions
+	x, y, width, height := n.GetInnerRect()
+	rightLimit := x + width
+	if height < 1 || rightLimit <= x {
+		return
+	}
+
+	// Draw label.
+	if n.labelWidth > 0 {
+		labelWidth := n.labelWidth
+		if labelWidth > rightLimit-x {
+			labelWidth = rightLimit - x
+		}
+		tview.Print(screen, n.label, x, y, labelWidth, tview.AlignLeft, n.labelColor)
+		x += labelWidth
+	} else {
+		_, drawnWidth := tview.Print(screen, n.label, x, y, rightLimit-x, tview.AlignLeft, n.labelColor)
+		x += drawnWidth
+	}
+
+	// get as much space as needed or available
+	fieldWidth := math.MaxInt32
+	if rightLimit-x < fieldWidth {
+		fieldWidth = rightLimit - x
+	}
+
+	// Draw the background
+	var fieldStyle tcell.Style
+	if n.HasFocus() {
+		fieldStyle = tcell.StyleDefault.Background(n.backgroundColorActivated)
+	} else {
+		fieldStyle = tcell.StyleDefault.Background(n.backgroundColor)
+	}
+
+	for index := 0; index < len(n.options[n.selectedOption]); index++ {
+		screen.SetContent(x+index, y, ' ', nil, fieldStyle)
+	}
+
+	// Draw the text
+	var textColor tcell.Color
+	if n.HasFocus() {
+		textColor = n.textColorActivated
+	} else {
+		textColor = n.textColor
+	}
+	text := n.options[n.selectedOption]
+	tview.Print(screen, tview.Escape(text), x, y, fieldWidth, tview.AlignLeft, textColor)
+}
+
+// InputHandler returns the handler for this primitive.
+func (n *EnumField) InputHandler() func(event *tcell.EventKey, setFocus func(p tview.Primitive)) {
+	return n.WrapInputHandler(func(event *tcell.EventKey, setFocus func(p tview.Primitive)) {
+		key := event.Key()
+		switch key {
+		case tcell.KeyLeft:
+			n.selectedOption--
+			if n.selectedOption < 0 {
+				n.selectedOption = len(n.options) - 1
+			}
+		case tcell.KeyRight:
+			n.selectedOption++
+			if n.selectedOption == len(n.options) {
+				n.selectedOption = 0
+			}
```

Comment:
You could make it shorted by making use of the modulo operator.

Reply:
Golang does not have a modulo operator.

Reply:
It does, it's `%`. I see I called it wrong, though, it's "modulus" - sorry for that.

Reply:
Discussed offline - I'm fine either way.